### PR TITLE
Add --info Option to Workload Base Command For Future Triage

### LINKED ISSUE #22148: Add additional details on the workload version in dotnet workload --list
### Is your feature request related to a problem? Please describe.
Customers want to know the version of what they have installed. Unfortunately, multiple packs live within a single workload and even that is tied to a manifest version as the workload itself doesn't have a version.  Likely needs PM guidance.  Should this be a separate parameter? Is manifest version enough?  Should it list out the version of every pack installed related to that workload?

Hypothesis: The manifest version is enough as that's used for rollback updates.

### Describe the solution you'd like
Add a column that shows the manifest version when adding an additional parameter to the list command (--verbosity).  Could include pack versions on verbosity diagnostics but that may not be a priority.

### PR DESCRIPTION
**Pull request by nagilson**
Part 2/2 for adding version info for workloads. (Other part was merged in to add this info to list.) 
Resolves https://github.com/dotnet/sdk/issues/22148#issuecomment-1168005944  See sample output here: 

Current Style after feedback:

image

OLD Style:

image

It also shows that using --info on sub commands doesn't work and the old error message still applies if --info is not provided.

Regarding the tests: Talking to @joeloff and I'll add a manual test because testing the actual output would require us to engineer many mock classes for something that is relatively static and unlikely to frequently change or be influenced by other code changes.

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs [hunk 1]
```diff
@@ -15,14 +20,68 @@ internal static class WorkloadCommandParser
...
         }
 
+        internal static void ShowWorkloadsInfo(IWorkloadInfoHelper workloadInfoHelper = null, IReporter reporter = null)
+        {
+            workloadInfoHelper ??= new WorkloadInfoHelper();
+            IEnumerable<WorkloadId> installedList = workloadInfoHelper.InstalledSdkWorkloadIds;
+            InstalledWorkloadsCollection installedWorkloads = workloadInfoHelper.AddInstalledVsWorkloads(installedList);
+            reporter ??= Cli.Utils.Reporter.Output;
+
+            if(!installedList.Any())
```
Comment (joeloff) [inline src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs:38]: missing space

File: src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs [hunk 2]
```diff
@@ -15,14 +20,68 @@ internal static class WorkloadCommandParser
...
+            {
+                reporter.WriteLine(CommonStrings.NoWorkloadsInstalledInfoWarning);
+                return;
+            }
+
+
+            foreach (var workload in installedWorkloads.AsEnumerable())
+            {
+                reporter.WriteLine("\n");
+                reporter.WriteLine(CommonStrings.WorkloadIdColumn + " : [" + workload.Key + "]");
```
Comment (joeloff) [inline src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs:48]: Extra space between the `:` and value on the left. 

image

File: src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs [hunk 3]
```diff
@@ -15,14 +20,68 @@ internal static class WorkloadCommandParser
...
+                var workloadManifest = workloadInfoHelper.WorkloadResolver.GetManifestFromWorkload(new WorkloadId(workload.Key));
+                var workloadFeatureBand = new WorkloadManifestInfo(
+                    workloadManifest.Id,
+                    workloadManifest.Version,
+                    Path.GetDirectoryName(workloadManifest.ManifestPath)!).ManifestFeatureBand;
+
+                reporter.Write(CommonStrings.WorkloadManfiestVersionColumn + ":");
+                reporter.WriteLine("\t" + workloadManifest.Version + "/" + workloadFeatureBand);
+
+                reporter.Write(CommonStrings.WorkloadManifestPathColumn + ":");
```
Comment (joeloff) [inline src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs:62]: I'm a bit on the fence with the inline tabs. Can we use interpolated strings with column widths or maybe a `PrintableTable` without column headings, that way the formatting remains more consistent, especially if the actual spaces in tabs varies between platforms

Comment (nagilson) [reply src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs:62]: I don't think we should use a `printabletable`, I used that initially, but:
a) it doesn't match `dotnet --info` even if it matches `dotnet workload list`
b) the manifest path is so long that the table becomes unreadable unless the command line is full screen and on a huge monitor

Interpolated strings though I advocate for!

Comment (nagilson) [reply src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs]: Please see committed fixes.

**Pull request merged**